modules/compute/gke-job-template/main.tf

```diff
@@ -61,9 +61,18 @@ locals {
     value = "true"
   }] : []
 
-  # arbitrarily, user can edit in template.
-  # May come from node pool in future.
-  gpu_limit_string = alltrue(var.has_gpu) ? "1" : null
+  # Start with the minimum gpu available across nodes in each pool
+  min_allocatable_gpu = min(var.allocatable_gpu_per_node...)
+  gpu_limit_string = (
+    alltrue(var.has_gpu) ?                  # if nodes in each pool are configured with GPUs
+    var.requested_gpu_per_pod > 0 ?         # if user supplied requested gpu
+    tostring(var.requested_gpu_per_pod) :   # then honor it
+    (                                       # else
+      local.min_allocatable_gpu > 0 ?       # if allocatable gpu was supplied
+      tostring(local.min_allocatable_gpu) : # then honor it
+      null                                  # else gpu limit is unset
+    ) : null                                # all node pools are configured without gpu, so gpu limit is unset
+  )
```

Comment:
Appreciate the effort to explain this out with a detailed comment. Can we instead break this expression into meaningful named smaller conditionals, and then hierarchically combine them?

Reply:
Rationale: let the code explain itself, this will need us to keep the code and comment in sync